hack/update-api-resource-list.sh

```diff
@@ -0,0 +1,110 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to fetch latest api resources served by kube apiserver
+# Puts the updated list at test/e2e/testing-manifests/apiresource/resources_all.csv
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+
+make -C "${KUBE_ROOT}" WHAT=cmd/kube-apiserver
+
+function cleanup()
+{
+    [[ -n ${APISERVER_PID-} ]] && kill ${APISERVER_PID} 1>&2 2>/dev/null
+
+    kube::etcd::cleanup
+
+    kube::log::status "Clean up complete"
+}
+
+function fetch_api_resource()
+{
+for RESOURCE in $(curl -s $APISERVER_URL/api/v1 | jq -c .resources[]); do
+  NAME=$(echo $RESOURCE | jq .name | tr -d '"')
+  NAMESPACED=$(echo $RESOURCE | jq .namespaced)
+  for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+    echo ,v1,$NAME,$NAMESPACED,$VERB
+  done
+done
+
+# List group versions
+for GROUP_VERSION in $(curl -s $APISERVER_URL/apis | jq .groups[].versions[].groupVersion | tr -d '"'); do
+  # List resources
+  for RESOURCE in $(curl -s $APISERVER_URL/apis/$GROUP_VERSION | jq -c '.resources[]'); do
+    NAME=$(echo $RESOURCE | jq '.name' | tr -d '"')
+    NAMESPACED=$(echo $RESOURCE | jq '.namespaced')
+    for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+      echo $(echo $GROUP_VERSION | tr '/' ','),$NAME,$NAMESPACED,$VERB
+    done
+  done
+done
+}
+
+trap cleanup EXIT SIGINT
+
+kube::golang::setup_env
+
+apiserver=$(kube::util::find-binary "kube-apiserver")
+
+TMP_DIR=$(mktemp -d /tmp/update-api-resource-list.XXXX)
+ETCD_HOST=${ETCD_HOST:-127.0.0.1}
+ETCD_PORT=${ETCD_PORT:-2379}
+API_PORT=${API_PORT:-8050}
+API_HOST=${API_HOST:-127.0.0.1}
+
+kube::etcd::start
+
+echo "dummy_token,admin,admin" > $TMP_DIR/tokenauth.csv
+
+# Start kube-apiserver
+kube::log::status "Starting kube-apiserver"
+"${KUBE_OUTPUT_HOSTBIN}/kube-apiserver" \
+  --insecure-bind-address="${API_HOST}" \
+  --bind-address="${API_HOST}" \
+  --insecure-port="${API_PORT}" \
+  --etcd-servers="http://${ETCD_HOST}:${ETCD_PORT}" \
+  --advertise-address="10.10.10.10" \
+  --cert-dir="${TMP_DIR}/certs" \
+  --runtime-config="api/all=true" \
+  --token-auth-file=$TMP_DIR/tokenauth.csv \
+  --logtostderr \
+  --v=2 \
+  --service-cluster-ip-range="10.0.0.0/24" >/tmp/openapi-api-server.log 2>&1 &
+APISERVER_PID=$!
+
+kube::util::wait_for_url "${API_HOST}:${API_PORT}/healthz" "apiserver: "
+
+APISERVER_URL=${API_HOST}:${API_PORT}
+
+KUBE_RESOURCE_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources_all.csv"
+KUBE_RESOURCE_WHITELIST_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources_whitelist.csv"
+KUBE_RESOURCE_TEST_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources.csv"
+
+# Make sure subresource come after parent resource
+fetch_api_resource | sort -t',' -k1,1 -k2,2 -k3,3 -k4 -o ${KUBE_RESOURCE_FILE}
```

Comment:
does this work with a BSD sort and grep on Mac?

Reply:
BSD sort and GNU sort give same result. BSD grep doesn't work well with -x flag. Switched to use `comm` which works consistently in both GNU and BSD